### ISSUE TITLE
Screen textures saved with wrong color

### ISSUE THREAD

**Issue by rossig7** [arturoc/ofxTextureRecorder#4]
Hi, I don't know what I am doing wrong. The settings are the same provided in the example.
These are the code I´ve been testing so far.
1- with fbo

```
	```
imgScreen.grabScreen(0, 0, ofGetWidth(), ofGetHeight());
	fbo.begin();
	    ofClear(0,0,0);			
	   imgScreen.draw(0, 0);
	fbo.end();
	recorder.save(fbo.getTexture());
```

2 - without using fbo
	
```
    imgScreen.grabScreen(0, 0, ofGetWidth(), ofGetHeight());
    recorder.save(imgScreen.getTexture());

```

However it works randomly in both cases, sometimes it works and sometimes it gives either images with black color or images with changed colors..like if it they were tinted.
The settings are the same provided in the example.
Thanks in advance for any hint.

OS: Windows 10
oF: 0.10
![frame_00001](https://user-images.githubusercontent.com/8951558/46180653-fc16c100-c297-11e8-830a-1e61f737d719.jpg)

**Comment by arturoc** [comment]
not sure what you are trying to do but the whole point of this addon is not having to use ofImage::grabScreen which is pretty slow.

Just draw your scene to an fbo and then save the fbo texture to with the recorder as it's done in the example

**Comment by rossig7** [comment]
Well, I need to do screen captures of my app. I run de app in a FULSCREEN mode. 
I did it using fbo in the first place as I pointed in the code above. But I think you meant to do the whole drawing in there without using grabScreen, right?. I will do.

Besides performance issues, is it possible that grabScreen generates some change in the channels?

**Comment by arturoc** [comment]
no idea, looks like you would be using rgba on one place and rgb on the other or something like that.

with screen grab you are downloading the whole screen from the graphics card into ram, uploading it again and then downloading it again with when you pass the ofImage texture to the recorder.

just use an fbo to draw your application, pass it's texture to the recorder and draw it to the screen as well

**Comment by rossig7** [comment]
Yes I got it. I changed the code, now everything is drawn into an fbo, and then I render from the fbo, I got the texture and so on. Everything is going well.

Just for reference: I no longer use grabScreen, even though, regarding this topic, what I changed it is mentioned in [this oF post](https://forum.openframeworks.cc/t/wrong-color-saved-when-using-ofimage-save-jpeg-after-using-ofimage-grabscreen-within-v0-10-0/30240), 
The only difference is that I used `auto pixelFormat = OF_PIXELS_RGBA;` instead.

Thanks a lot for your help!
